homeassistant/components/influxdb/sensor.py

```diff
@@ -139,7 +153,8 @@ def validate_query_format_for_version(conf: dict) -> dict:
     SENSOR_PLATFORM_SCHEMA.extend(COMPONENT_CONFIG_SCHEMA_CONNECTION).extend(
         {
             vol.Exclusive(CONF_QUERIES, "queries"): [_QUERY_SCHEMA[LANGUAGE_INFLUXQL]],
-            vol.Exclusive(CONF_QUERIES_FLUX, "queries"): [_QUERY_SCHEMA[LANGUAGE_FLUX]],
+            vol.Exclusive(CONF_QUERIES_RAW, "queries"): [_QUERY_SCHEMA[LANGUAGE_INFLUXQL_RAW]],
```

Comment:
I don't think `queries_raw` should be exclusive here. By defining it this way you make it so users must specify all their queries in raw or none of their queries in raw. Why wouldn't a user be able to mix and match? Specify one complex query as raw but the others as normal using `queries` or `queries_flux`? I feel like the schema should only require that `CONF_QUERIES` and `CONF_QUERIES_FLUX` are exclusive since an Influx deployment can only accept queries in one language. `CONF_QUERIES_RAW` should be be separate and not exclusive with either.

The schema should also validate that at least one of `CONF_QUERIES`, `CONF_QUERIES_FLUX` and `CONF_QUERIES_RAW` is provided.

Reply:
I agree that it's better to be able to mix... I'm looking for improving the proposed changes on that point.

Is there a proper way to do the the validation of configuration if set to optional (ie QUERIES and QUERIES_FLUX cannot be set together)?

Reply:
Yea just don't bundle the new one into the `exclusive`. `vol.Exclusive` means that one one of the things in the named exclusive (`queries` in this case) can exist. Since `queries_raw` can co-exist with the others you just make it a normal `vol.Optional`. So it would be like this:

```python
    SENSOR_PLATFORM_SCHEMA.extend(COMPONENT_CONFIG_SCHEMA_CONNECTION).extend(
        {
            vol.Exclusive(CONF_QUERIES, "queries"): [_QUERY_SCHEMA[LANGUAGE_INFLUXQL]],
            vol.Optional(CONF_QUERIES_RAW): [_QUERY_SCHEMA[LANGUAGE_INFLUXQL_RAW]],
            vol.Exclusive(CONF_QUERIES_FLUX, "queries"): [_QUERY_SCHEMA[LANGUAGE_FLUX]],            
        }
    ),
```

Reply:
Taken into account in the e7dfb9f commit